### DISCUSSION TITLE
[Question]: How I can change default port 3000?

### DISCUSSION THREAD

**Discussion by IsOriginalNickname** [danny-avila/LibreChat#4221] [Q&A]
### What is your question?

I am trying to deploy LibreChat on my server using Dokploy. Unfortunately, port 3000 is already in use, and I cannot find where to change the port configuration. 

Could you please guide me on how to modify the port settings for LibreChat? 

### More Details

I've got error where I try to deploy LibreChat with default config:
**Bind for 0.0.0.0:3000 failed: port is already allocated**

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by kukoboris** [comment]
Look at **.env** and **docker-compose.yml** or **deploy-compose.yml** (depending on which installation method you use)

**Reply by IsOriginalNickname** [reply]
I've tried. There is no port 3000 in any of them.

My docker compose file has next:
    container_name: LibreChat-API
    ports:
      - 3080:3080
      
And .env has the same port on all config lines.

**Reply by structwafel** [reply]
I have the same issue.

port 3000 is in use by another docker container on the network.

all the config in librechat is to point to something else like 3080.
but something on startup is trying to bind to 3000.

nothing in the repo suggests listening to 3000

**Reply by danny-avila** [reply]
The only thing that could cause this if if there is something in your environment declaring a value for `PORT`

If the `PORT` environment variable is set elsewhere in your system, it could override the value in your .env file. Here's how you can check the value of the `PORT` environment variable on different operating systems:

1. Linux and macOS:
   Open a terminal and run:
   ```
   echo $PORT
   ```
   Or to see all environment variables:
   ```
   env | grep PORT
   ```

2. Windows (Command Prompt):
   Open Command Prompt and run:
   ```
   echo %PORT%
   ```
   Or to see all environment variables:
   ```
   set | findstr PORT
   ```

3. Windows (PowerShell):
   Open PowerShell and run:
   ```
   $env:PORT
   ```
   Or to see all environment variables:
   ```
   Get-ChildItem Env: | Where-Object { $_.Name -like "*PORT*" }
   ```

If these commands return a value, it means the `PORT` variable is set in your system environment. If they don't return anything, the variable is not set at the system level.

To temporarily unset the variable for your current session:

- On Linux/macOS:
  ```
  unset PORT
  ```

- On Windows (Command Prompt):
  ```
  set PORT=
  ```

- On Windows (PowerShell):
  ```
  Remove-Item Env:PORT
  ```

After unsetting the variable, you can try running your Docker commands again. If the issue persists, you might want to explicitly set the `PORT` in your .env file or in the docker-compose command:

```
PORT=3001 docker-compose up -d
```

This will override any system-level environment variable for this specific command.

**Comment by RemcoE33** [comment]
@IsOriginalNickname did you get Librechat itself working? If so, would love to know the setup.
